### PR TITLE
[WF-494] User Account menu

### PR DESCRIPTION
**Pull request by blurbyte**
# [WF-494](https://datacamp.atlassian.net/browse/WF-494)

## Proposed changes

Implemented **accessible menu** according to WCAG guidelines: https://www.w3.org/TR/wai-aria-practices-1.1/#menu
Main features:
- supports full keayboard and mouse navigation with proper _wai-aria_ attributes
- provides fallback user profile image if it fails to load
- it's possible to extend it with custom menu items (`UserAccountMenu.MenuItem`)
- hides _My Profile_ and _XP_ sections if appropriate props are not passed
- it's possible to adjust its vertical position with `dropdownOffset`
- custom track IDs could be passed to menu elements if needed

Interactions:
user-profile-menu

## Functional Code

- [x] Builds without errors
- [x] Linter passes CI
- [x] Unit tests written & pass CI
- [x] Integration tests written & pass CI
- [x] Tested on [supported browsers](https://support.datacamp.com/hc/en-us/articles/360001541574-Minimum-System-Requirements)

## Documentation

- [x] Technical docs written
- [x] Structural changes reflected in Readme
- [ ] ~~Migration plan for breaking changes~~

## Meets Product Requirement

- [x] Assumptions are met
- [x] Meets acceptance criteria
- [x] Approved by Designer, Engineer, & PO

### CODE REVIEW

File: packages/react-components/button/src/UserAccountMenu/Avatar.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { css } from '@emotion/react';
+import { useState } from 'react';
+
+import { FULL_MENU_MEDIA_QUERY } from './constants';
+import defaultAvatar from './defaultAvatar';
+
+const wrapperStyle = css({
+  borderRadius: '50%',
+  [FULL_MENU_MEDIA_QUERY]: {
```
Comment (dandenney) [inline packages/react-components/button/src/UserAccountMenu/Avatar.tsx:9]: This is interesting, I hadn't seen it handled this way before

**Review by dandenney — APPROVED** [review]
This looks great! I don't have anything constructive to offer, it's a higher quality component than I likely would have put out. It might make for a great knowledge share, in stepping through some of the decisions like the media query, how to work with react-aria, etc

**Comment by blurbyte** [comment]
@dandenney Thanks. I think the idea of knowledge share sounds great. I will come up with something in July.

**Comment by dandenney** [comment]
> @dandenney Thanks. I think the idea of knowledge share sounds great. I will come up with something in July.

Great! And maybe you can do the new mega nav when the design is done :trollface:

**Pull request merged**